### PR TITLE
add support for DatasetTemplate resource

### LINKED ISSUE #88: Add support for DatasetTemplate resource

### PR DESCRIPTION
**Pull request by davecap**
closes #88

### CODE REVIEW

File: R/dataset_template.R [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+#' DatasetTemplate.all()
+#' }
+#'
+#' @references
+#' \url{https://docs.solvebio.com/}
+#'
+#' @export
+DatasetTemplate.all <- function(env = solvebio:::.solveEnv, ...) {
+    .request('GET', "v2/dataset_templates", query=list(...), env=env)
+}
```
Comment (jsh2134) [inline R/dataset_template.R]: can't seem to get the is_recipe flag to work, but maybe it's my ignoRance
```
> DatasetTemplate.all()$total
[1] 2326
> DatasetTemplate.all(tags="import")$total
[1] 7
> DatasetTemplate.all(is_recipe=TRUE)$total
[1] 2326
```

File: R/dataset_template.R [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+#'
+#' @export
+DatasetTemplate.retrieve <- function(id, env = solvebio:::.solveEnv) {
+    if (missing(id)) {
+        stop("A dataset template ID is required.")
+    }
+
+    path <- paste("v2/dataset_templates", paste(id), sep="/")
+    .request('GET', path=path, env=env)
+}
```
Comment (jsh2134) [inline R/dataset_template.R]: works 👍

File: R/dataset_template.R [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+#' @export
+DatasetTemplate.delete <- function(id, env = solvebio:::.solveEnv) {
+    if (missing(id)) {
+        stop("A dataset template ID is required.")
+    }
+
+    path <- paste("v2/dataset_templates", paste(id), sep="/")
+    .request('DELETE', path=path, env=env)
+}
+
```
Comment (jsh2134) [inline R/dataset_template.R]: works 👍

File: R/dataset_template.R [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+#' DatasetTemplate.create(name="My Dataset Template")
+#' }
+#'
+#' @references
+#' \url{https://docs.solvebio.com/}
+#'
+#' @export
+DatasetTemplate.create <- function(env = solvebio:::.solveEnv, ...) {
+    .request('POST', path='v2/dataset_templates', query=NULL, body=list(...), env=env)
+}
```
Comment (jsh2134) [inline R/dataset_template.R]: works 👍

File: R/dataset_template.R [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+#' \url{https://docs.solvebio.com/}
+#'
+#' @export
+DatasetTemplate.update <- function(id, env = solvebio:::.solveEnv, ...) {
+    if (missing(id)) {
+        stop("A dataset template ID is required.")
+    }
+
+    path <- paste("v2/dataset_templates", paste(id), sep="/")
+    .request('PATCH', path=path, query=NULL, body=list(...), env=env)
```
Comment (jsh2134) [inline R/dataset_template.R]: works 👍

File: R/dataset_template.R (hunk 1 above)
Comment (davecap) [reply R/dataset_template.R]: `DatasetTemplate.all(is_recipe="True")$total`

This is an API issue (filed as a separate ticket).

**Pull request merged**